src/clients/FlexbaseClient.Card.ts

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Card } from '../models/Card/Card';
+import { Address } from '../models/Address/Address';
+import { CardOptions } from '../models/Card/CardOptions';
+import { FlexbaseClientBase } from './FlexbaseClient.Base';
+import { FlexbaseResponse } from '../models/FlexbaseResponse';
+
+interface QueryParameters {
+    searchTerm?: string;
+    status?: string;
+}
+
+interface CardsResponse extends FlexbaseResponse{
+    cards: Card[];
+}
+
+interface CardResponse extends FlexbaseResponse{
+    card: Card;
+}
+
+interface IssueCardForm {
+    cardType: string,
+    shipTo?: Address,
+    service?: string,
+}
+
+interface UpdateCardForm {
+    expensesTypes: {
+        amount: number,
+        groups: any,
+    };
+    notifyUse: boolean;
+    shipTo?: Address,
+    creditLimit: number,
+    cardName?: string,
+}
+
+export class FlexbaseClientCard extends FlexbaseClientBase {
+    private params({ searchTerm, status }: CardOptions = {}) {
+        const params: any = {};
+
+        if (typeof searchTerm === 'string') {
+            params.searchTerm = searchTerm;
+          }
+      
+          if (typeof status === 'string') {
+            params.status = status;
+          }
+
+        return params;
+    }
+
+    async getCompanyCards({ searchTerm, status }: QueryParameters = {}): Promise<Card[] | null> {
+        try {
+          const params = this.params({ searchTerm, status });
+    
+          const response = await this.client.url('/card/company').query(params).get().json<CardsResponse>();
+
+           if (!response.success) {
+              return null;
+           }
+    
+          return response.cards;
+        } catch (error) {
+          this.logger.error('Unable to get company cards', error);
+          return null;
```

Comment:
Is returning `null` what we want here for erroneous responses? Wouldn't an empty array `[]` be more fitting to send forward an empty list of cards for consumption?

@robert-dean @kyle-b2 @peilong-du Furthermore, for error-handling, upon catching an error, shouldn't we be sending more info/context around the error back to the consumer of this call? Perhaps, a message similar to the one that we are logging?

Reply:
@huy-flex responded below about the return but when presenting errors back to the user we need to be very careful what is shown. From a security practice its best to provide generic messaging to the user and log internally the details.

Reply:
See follow-up below.